Dump states

sbeaver-netgate opened the pull request:
Incorporates changes to:
diag_dump_states.php
diag_dump_states_sources.php
diag_reset_state.php

sbeaver-netgate committed:
dig_dump_states related pages updated to bootstrap

sbeaver-netgate committed:
Tabs/spaces

sbeaver-netgate committed:
Updated icons

Icons replaces with xs buttons

sbeaver-netgate committed:
Remove temporary comment block

sbeaver-netgate committed:
Corrected typo (trid)

sbeaver-netgate committed:
Replaced icon with btn-xs

SanderVanLeeuwen commented on usr/local/www/diag_dump_states.php:
```diff
@@ -139,42 +132,36 @@ function(index,row) { jQuery(row).fadeOut(1000); }
+	</div>
+
+	</form>
+	<br>
```
Please remove this and other `<br>`'s which are only used for spacing. That's _not_ what this element is for.

sbeaver-netgate replied:
Done

> On Apr 22, 2015, at 4:40 PM, Sander van Leeuwen notifications@github.com wrote:
> 
> In usr/local/www/diag_dump_states.php https://github.com/SjonHortensius/pfsense/pull/97#discussion_r28912524:
> 
> > -   <?=gettext("&nbsp;Current total state count")?>: <?= $current_statecount ?>
> > -   <div class="pull-right">
> > -       <?=gettext("Filter expression:")?>
> > -       <input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_POST['filter'])?>" size="30" />
> > -       <input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
> >   +
> >   +<?php
> > -       if (isset($_POST['filter']) && (is_ipaddr($_POST['filter']) || is_subnet($_POST['filter'])))
> > -          print('<input type="submit" class="btn btn-danger btn-sm formbtn" name="killfilter" value="Kill">');
> >   +?>
> >   +
> > -   &nbsp;
> > -   </div>
> >   +
> > -   </form>
> > -   <br>
> >   Please remove this and other <br>'s which are only used for spacing. That's not what this element is for.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/97/files#r28912524.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

sbeaver-netgate committed:
Removed bogus &nbsp;

sbeaver-netgate committed:
Revised conversion

HTML filter controls replaced with Form class
Outer panel removed
Ready for review

sbeaver-netgate committed:
Update revised

Eliminated all HTML

SjonHortensius commented on usr/local/www/diag_dump_states.php:
```diff
@@ -54,11 +54,11 @@
 	return;
 }
 
-if (isset($_POST['filter']) && isset($_POST['killfilter'])) {
```
Could you revert this and rename the input back to the original 'filter'?

SjonHortensius commented on usr/local/www/diag_dump_states.php:
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
+	['placeholder' => '']
+));
+
+$section->addInput(new Form_StaticText(
```
Why don't you move the Button to the Form-constructor? I'm curious

SjonHortensius commented on usr/local/www/diag_dump_states.php:
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
+	'filtertext',
+	'Filter expression',
+	'text',
+	htmlspecialchars($_POST['filtertext']),
```
htmlspecialchars not needed here

SjonHortensius commented on usr/local/www/diag_dump_states.php:
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
-	$parts = explode(":", $ends[count($ends) - 1]);
-	$dstip = trim($parts[0]);
+<script>
+	document.getElementById("filter").className = "btn btn-default btn-sm";
```
Why not do this in the php-code?

SjonHortensius commented on usr/local/www/diag_dump_states.php:
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
+<table class="table table-striped table-hover table-compact" summary="results">
+	<thead>
+		<tr class="info">
+			<th class="col-md-1"><?=gettext("Int")?></th>
```
I think these classes aren't required

SjonHortensius commented on usr/local/www/diag_dump_states.php:
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
-		</td>
-	</tr>
+		<tr>
+			<td colspan="4" align="center">
```
Please remove the align too. Maybe wrap the message in an alert instead?

SjonHortensius commented on usr/local/www/diag_dump_states_sources.php:
```diff
@@ -104,54 +102,41 @@ function(index,row) { jQuery(row).fadeOut(1000); }
 
 <!-- Start of tab content -->
+	<br />
+	<form action="<?=$_SERVER['SCRIPT_NAME']?>" method="get">
```
Is the reason you don't use a proper form here, that you want to use GET?

SjonHortensius commented on usr/local/www/diag_dump_states_sources.php:
```diff
@@ -104,54 +102,41 @@ function(index,row) { jQuery(row).fadeOut(1000); }
+			<?=gettext("Filter expression:")?>
+			<input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_GET['filter'])?>" size="30" />
+			<input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
+			&nbsp;
```
not required

SjonHortensius commented on usr/local/www/diag_dump_states_sources.php:
```diff
@@ -104,54 +102,41 @@ function(index,row) { jQuery(row).fadeOut(1000); }
+	<table class="table table-striped table-hover table-compact">
+		<thead>
+			<tr>
+				<th class="listhdrr" width="40%"><?=gettext("Source -> Destination")?></th>
```
all these classes are not required (the widths neither)

sbeaver-netgate replied:
Sjon,

This looks like an old version.

Standby while I check please. There is no way I left all that junk in these files.

Steve

> On Apr 28, 2015, at 2:44 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dump_states_sources.php https://github.com/SjonHortensius/pfsense/pull/97#discussion_r29274787:
> 
> >  <!-- Start of tab content -->
> > -   <br />
> > -   <form action="<?=$_SERVER['SCRIPT_NAME']?>" method="get">
> > -       <div class="pull-right">
> > -           <?=gettext("Filter expression:")?>
> > -           <input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_GET['filter'])?>" size="30" />
> > -           <input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
> > -           &nbsp;
> > -       </div>
> > -   </form>
> > -   <br>
> > -   <div class="responsive">
> > -   <table class="table table-striped table-hover table-compact">
> > -       <thead>
> > -           <tr>
> > -               <th class="listhdrr" width="40%"><?=gettext("Source -> Destination")?></th>
> >   all these classes are not required (the widths neither)
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/97/files#r29274787.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

SjonHortensius replied:
I hope, I was pretty surpised as well. I'll have another look at the diff

On Tue, Apr 28, 2015 at 8:46 PM, Steve Beaver notifications@github.com
wrote:

> In usr/local/www/diag_dump_states_sources.php
> https://github.com/SjonHortensius/pfsense/pull/97#discussion_r29274970:
> 
> >  <!-- Start of tab content -->
> > -   <br />
> > -   <form action="<?=$_SERVER['SCRIPT_NAME']?>" method="get">
> > -       <div class="pull-right">
> > -           <?=gettext("Filter expression:")?>
> > -           <input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_GET['filter'])?>" size="30" />
> > -           <input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
> > -           &nbsp;
> > -       </div>
> > -   </form>
> > -   <br>
> > -   <div class="responsive">
> > -   <table class="table table-striped table-hover table-compact">
> > -       <thead>
> > -           <tr>
> > -               <th class="listhdrr" width="40%"><?=gettext("Source -> Destination")?></th>
> 
> Sjon, This looks like an old version. Standby while I check please. There
> is no way I left all that junk in these files. Steve
>  On Apr 28, 2015, at 2:44 PM, SjonHortensius notifications@github.com
> wrote: In usr/local/www/diag_dump_states_sources.php <
> https://github.com/SjonHortensius/pfsense/pull/97#discussion_r29274787>:
> 
> > > <!-- Start of tab content --> > + <br /> > + <form
> > > action="<?=$_SERVER['SCRIPT_NAME']?>" method="get"> > + <div
> > > class="pull-right"> > + <?=gettext("Filter expression:")?> > + <input
> > > type="text" name="filter" class="formfld search"
> > > value="<?=htmlspecialchars($_GET['filter'])?>" size="30" /> > + <input
> > > type="submit" class="btn btn-primary btn-sm formbtn"
> > > value="<?=gettext("Filter")?>" /> > + &nbsp; > + </div> > + </form> > +
> > > <br> > + <div class="responsive"> > + <table class="table table-striped
> > > table-hover table-compact"> > + <thead> > + <tr> > + <th class="listhdrr"
> > > width="40%"><?=gettext("Source -> Destination")?></th> all these classes
> > > are not required (the widths neither) — Reply to this email directly or
> > > view it on GitHub <
> > > https://github.com/SjonHortensius/pfsense/pull/97/files#r29274787>.
> > > Steve -- Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com
> > > mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SjonHortensius/pfsense/pull/97/files#r29274970.

SjonHortensius commented on usr/local/www/diag_dump_states_sources.php:
```diff
@@ -169,44 +154,37 @@ function(index,row) { jQuery(row).fadeOut(1000); }
-		<?php
+?>
+        	<tr name='r:<?="{$srcip}:{$dstip}" ?>'>
+        		<td class='listlr'><?=$info?></td>
```
These classes are not needed. Pretty sure the tr.name isn't either

SjonHortensius commented on usr/local/www/diag_dump_states_sources.php:
```diff
@@ -169,44 +154,37 @@ function(index,row) { jQuery(row).fadeOut(1000); }
-		  </tr>";
+?>
+	        <tr>
+        		<td class="list" colspan="4" align="center" valign="top">
```
please remove class, align & valign here

SjonHortensius commented on usr/local/www/diag_resetstate.php:
```diff
@@ -65,73 +75,65 @@
+
+$form = new Form(new Form_Button(
+	'Submit',
+	gettext('Reset')
```
gettext not needed in calls to Form_*

sbeaver-netgate commented:
PR and feature branched closed. A local restore from backup caused these files to become out of date.
Three new PRs will be submitted for each of the files
